src/structures/APIMessage.js

```diff
@@ -151,8 +151,8 @@ class APIMessage {
 
     let nonce;
     if (typeof this.options.nonce !== 'undefined') {
-      nonce = parseInt(this.options.nonce);
-      if (isNaN(nonce) || nonce < 0) throw new RangeError('MESSAGE_NONCE_TYPE');
+      nonce = typeof this.options.nonce === 'string' ? this.options.nonce : parseInt(this.options.nonce);
+      if (isNaN(nonce) && !nonce) throw new RangeError('MESSAGE_NONCE_TYPE');
```

Comment:
why was the check for a negative nonce removed?
should this check also be a bit more explicit, `typeof nonce === 'number' && ...`?

Reply:
nonce could be negative. It just should be a integer

Reply:
ah right, what about a check for nonce being `0`, sending the nonce as 0 seems to be ignored by the api

Reply:
Looks like its working

Reply:
Made a change for new commit

Reply:
![20200903_155310](https://user-images.githubusercontent.com/28943913/92131351-ca042700-edfd-11ea-846c-74d748b02320.jpg) 0 nonce doesn't seem to be sent back

Reply:
I have checked by sending POST request to direct Discord API and it worked.

Reply:
Please check if i'm wrong!